### ISSUE TITLE
[Bug]: start_offset and end_offset set to NULL in refresh policies do not takes min() and max() of the hypertable

### ISSUE THREAD

**Issue by camsimi** [timescale/timescaledb#6800]
### What type of bug is this?

Configuration

### What subsystems and features are affected?

Continuous aggregate, Policy

### What happened?

I use a refresh policy on a continuous aggregate.

Reading the doc [here](https://docs.timescale.com/api/latest/continuous-aggregates/add_continuous_aggregate_policy/) I thought I could use `null` for `start_offset` and `end_offset` because it says that it would take the `MIN(timestamp)` and the `MAX(timestamp)` of the hypertable.

> Start of the refresh window as an interval relative to the time when the policy is executed. NULL is equivalent to MIN(timestamp) of the hypertable.

Instead, when the refresh policy is executed, I get this window : [ **4714**-11-24 00:00:00 BC, **294247**-01-02 00:00:00 ]

The data in the based hypertable and in the c-agg:
- starts on `2021-01-30 00:00:00.000000`, nothing before.
- ends on `2024-02-05 00:00:00.000000`.

The type of the timestamp field in the c-agg is timestamp without time zone. I've tried with a time field with TIMESTAMPTZ and reproduce the same issue . 

NB: I am aware of the warning about using a timestamp without timezone for the time field. I decide to use it like that consciously and thoughtfully.

### TimescaleDB version affected

2.14.2

### PostgreSQL version used

14.11.1

### What operating system did you use?

Ubuntu 22.4 x64

### What installation method did you use?

Docker

### What platform did you run on?

Managed Service for TimescaleDB (MST/Aiven), On prem/Self-hosted

### Relevant log output and stack trace

```bash
LOG:  refreshing continuous aggregate "hourly_consumptions" in window [ 4714-11-24 00:00:00 BC, 294247-01-02 00:00:00 ]
```

### How can we reproduce the bug?

```bash
-- Create measure hypertable with primary keys
CREATE TABLE measures (
    entity_id BIGINT,
    at TIMESTAMP,
    value BIGINT
);
ALTER TABLE measures ADD PRIMARY KEY (entity_id, at);
SELECT create_hypertable('measures', 'at');

-- Insert data
INSERT INTO measures VALUES (1, '2021-01-01 00:00:00', 10);
INSERT INTO measures VALUES (1, '2024-03-31 00:00:00', 20);

-- Create hourly consumptions continuous aggregate from measures hypertable
CREATE MATERIALIZED VIEW hourly_consumptions WITH (timescaledb.continuous) AS
SELECT
    entity_id,
    time_bucket('1 hour', at) AS hour,
    sum(value) AS consumption
FROM measures
GROUP BY hour, entity_id;

-- Create refresh policy for continuous aggregate
SELECT add_continuous_aggregate_policy(
        'hourly_consumptions',
        start_offset => NULL,
        end_offset => NULL,
        schedule_interval => INTERVAL '1 hour',
        initial_start => '2024-03-31 00:00:00'::timestamp
       );
```

**Comment by nikkhils** [comment]
@camsimi as per the above repro script, there's no data in the hypertable, right? In that case, MIN_POSSIBLE_TIMESTAMP and MAX_POSSIBLE_TIMESTAMP will be the min and max values, no?

**Comment by nikkhils** [comment]
If I insert data into the hypertable and run the refresh then I do see it pick up the max(at) value. 

```
2024-04-03 12:29:37.404 IST [3941] DEBUG:  Executing policy_refresh_continuous_aggregate with parameters {"end_offset": null, "start_offset": null, "mat_hypertable_id": 276}
2024-04-03 12:29:37.404 IST [3941] DEBUG:  initializing debug point 'hypertable_expansion_before_lock_chunk' to use 901696854
2024-04-03 12:29:37.404 IST [3941] CONTEXT:  SQL statement "SELECT pg_catalog.max(at) FROM public.measures"
```

**Comment by camsimi** [comment]
Sorry @nikkhils I forgot to put the data insertion script. I've just added it now.

When creating the the refresh policy I get this log:
`LOG:  refreshing continuous aggregate "hourly_consumptions" in window [ 4714-11-24 00:00:00 BC, 294247-01-02 00:00:00 ]`

And when running the job manually, I get the exact same log.

What should I do to enable the same DEBUG and CONTEXT logs as you have ?

**Comment by nikkhils** [comment]
@camsimi to test this I logged into `psql` and then ran the below:

```sql
set log_min_messages to debug3;
CALL run_job(:refreshjob_id);  -- you can get the refreshjob_id from timescaledb_information.jobs
```

**Comment by camsimi** [comment]
Thanks for the setting @nikkhils !

When calling the refresh job, I get these logs instead:

```
DEBUG:  parse S_1: 
bind <unnamed> to S_1
bind <unnamed> to S_1
bind <unnamed> to S_1
parse <unnamed>: SHOW TRANSACTION ISOLATION LEVEL
bind <unnamed> to <unnamed>
parse <unnamed>: CALL run_job(1043)
bind <unnamed> to <unnamed>
DEBUG:  Executing policy_refresh_continuous_aggregate with parameters {"end_offset": null, "start_offset": null, "mat_hypertable_id": 21}
LOG:  refreshing continuous aggregate "hourly_consumptions" in window [ 4714-11-24 00:00:00 BC, 294247-01-02 00:00:00 ]
STATEMENT:  CALL run_job(1043)
DEBUG:  hypertable 20 existing watermark >= new invalidation threshold 1711846800000000 1711846800000000
DEBUG:  parse <unnamed>: select current_database() as a, current_schemas(false) as b
DEBUG:  bind <unnamed> to <unnamed>
DEBUG:  parse <unnamed>: SHOW TRANSACTION ISOLATION LEVEL
DEBUG:  bind <unnamed> to <unnamed>
```

Does it help ?

**Comment by jnidzwetzki** [comment]
Hi @camsimi,

Thanks for getting back. Our documentation regarding these parameters is not precise. I have opened a [PR](https://github.com/timescale/docs/pull/3128) to update our documentation. When no `start_offset` and `end_offset` are specified, these values are taken from the invalidation log. However, no invalidation log exists when the CAgg is refreshed for the first time, and the whole time range is refreshed.

After the initial refresh is done, proper time ranges are picked up from the invalidation log:

```
test2=# INSERT INTO measures VALUES (2, '2024-03-31 00:00:00', 20);
test2=#  CALL run_job(1000);


2024-04-09 12:00:16.192 CEST [111806] LOG:  continuous aggregate refresh (individual invalidation) on "hourly_consumptions" in window [ 2024-03-31 00:00:00, 2024-03-31 01:00:00 ]
```

**Comment by camsimi** [comment]
@jnidzwetzki in your PR, you have updated Documentation about [`refresh_continuous_aggregate()` ](https://docs.timescale.com/api/latest/continuous-aggregates/refresh_continuous_aggregate/)(`window_start`,...) but not the one I have linked for the function [`add_continuous_aggregate_policy`()](url) I use (`start_offset`, ...).

And when calling the job for the second time, I still have the same largest window (not based on the timestamps I have in my hypertable).

**Comment by jnidzwetzki** [comment]
Hi @camsimi,

It is a known problem in TimescaleDB 2.14.x that the configured but not the actual refreshed range is logged. This is changed in the following PR (https://github.com/timescale/timescaledb/pull/6756), which will be part of 2.15.0. Could you update to TSDB 2.15.0 after it is released and check if this entirely solves the problem in your environment?

**Comment by camsimi** [comment]
Ok thanks for the information @jnidzwetzki 
As we're using Aiven Managed Service, it could take a while for the 2.15.0 to be available but I'll sure keep you informed !

**Comment by camsimi** [comment]
@jnidzwetzki all fixed with 2.15.0 indeed, thanks :)
